sdk/cosmos/azure-cosmos/samples/access_cosmos_with_resource_token.py

```diff
@@ -0,0 +1,235 @@
+            "permissionMode": documents.PermissionMode.All,
+            "resource": container.container_link,
+        }
+
+        permission = create_permission_if_not_exists(user, permission_definition)
+        token = {} 
+        token[container.id] = permission.properties["_token"] 
+
+        # Use token to connect to database
+        token_client = cosmos_client.CosmosClient(HOST, token)
+        token_db = token_client.get_database_client(DATABASE_ID)
+        token_container = token_db.get_container_client(CONTAINER_ID)
+
+        ITEM_1_ID, ITEM_2_ID, ITEM_3_ID  = "1", "2", "3"
+
+        # Update or insert item if not exists
+        token_client_upsert(token_container, USERNAME, ITEM_1_ID)
+        token_client_upsert(token_container, USERNAME, ITEM_2_ID)
+        token_client_upsert(token_container, USERNAME_2, ITEM_3_ID)
+
+        # Read all items in the container, across all partitions
+        token_client_read_all(token_container)
+
+        # Read specific item
+        token_client_read_item(token_container, USERNAME, ITEM_2_ID)
+
+        # Query for items in a certain partition
+        token_client_query(token_container, USERNAME_2)
+
+        # Delete an item
+        token_client_delete(token_container, USERNAME, ITEM_2_ID)
+
+    
+        # Give user read-only permission, for a specific partition 
+        user_2 = create_user_if_not_exists(db, USERNAME_2)
+        permission_definition = {
+            "id": PARTITION_READ_PERMISSION,
+            "permissionMode": documents.PermissionMode.Read,
+            "resource": container.container_link,
+            "resourcePartitionKey": [USERNAME_2]
+        }
+        permission = create_permission_if_not_exists(user_2, permission_definition)
+        read_token = {}
+        read_token[container.id] = permission.properties["_token"]
+ 
+        # Use token to connect to database
+        token_client = cosmos_client.CosmosClient(HOST, read_token)
+        token_db = token_client.get_database_client(DATABASE_ID)
+        token_container = token_db.get_container_client(CONTAINER_ID)
+
+        # Fails since this client has access to only items with partition key USERNAME_2 (ie. "user2")
+        token_client_read_all(token_container)
+
+        # Ok to read item(s) with partition key "user2"
+        token_client_read_item(token_container, USERNAME_2, ITEM_3_ID)
+
+        # Can't upsert or delete since it's read-only
+        token_client_upsert(token_container, USERNAME_2, ITEM_3_ID)
+
+
+        # Give user CRUD permissions, only for a specific item
+        user_2.delete_permission(PARTITION_READ_PERMISSION)
+        assert len(list(user_2.list_permissions())) == 0
+
+        permission_definition = {
+            "id": DOCUMENT_ALL_PERMISSION,
+            "permissionMode": documents.PermissionMode.All,
+            "resource": "dbs/FCJ5AA==/colls/FCJ5APInZmE=/docs/FCJ5APInZmELAAAAAAAAAA==/" #this identifies the item with id "3"
```

Comment:
If another user runs this sample, will they have different RIDs (resource ids) for database and collection? To make this more reproducable, it may be better to read the RID of the item with id 3 on the fly and use that in this permission_definition.

Reply:
I'll try to do that. It didn't seem to work before but I can look into it.